### PR TITLE
fix(nodejs): wait to write all data to disk before calling back

### LINKED ISSUE #8541: Node.js binding on Windows doesn't seem to release a file handle
### What happens?

I connect to a same duckDB file multiple times with Node.js binding on Windows. 
The second connection always fails due to the following error. It can be reproduced even if I connect in READONLY mode.

```
[Error: IO Error: Cannot open file "test.db": The process cannot access the file because it is being used by another process.
] {
   errno: -1,
   code: 'DUCKDB_NODEJS_ERROR',
   errorType: 'IO'
}
```

I know a similar issue has been found in DuckDB R package. `Database.close()` method of Node.js binding has no option which equivalent to the `SHUTDOWN=true` option and cannot be resolved by the solution of this issue.

https://github.com/duckdb/duckdb/issues/5481

### To Reproduce

It occurs when the following program is executed with Node.js on windows.

```js
var duckdb = require("duckdb");

const create = (dbPath) =>
  new Promise((resolve, reject) => {
    const db = new duckdb.Database(dbPath, 2, (err) => {
      if (err) reject(err);
      resolve(db);
    });
  });

const close = (db) =>
  new Promise((resolve, reject) => {
    db.close((err) => {
      if (err) reject(err);
      resolve();
    });
  });

(async () => {
  // first connection
  var db1 = await create("test.db");
  var con1 = db1.connect();
  console.log(con1);
  await close(db1);

  // second connection
  var db2 = await create("test.db");
  var con2 = db2.connect();
  console.log(con2);
  await close(db2);
})();
```

### OS:

Windows Server 2019 (Version 1809 OS Build 17763.3046) AMD64

### DuckDB Version:

v0.8.1, v0.8.2-dev2700

### DuckDB Client:

Node.js binding

### Full Name:

Yoshiki Kurihara

### Affiliation:

Yuimedi. inc

### Have you tried this on the latest `master` branch?

I have tested with a master build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

### PR DESCRIPTION
**Pull request by Mause**
Fixes https://github.com/duckdb/duckdb/issues/8541

Also restores support for CMake for the NodeJS bindings

Can remove some of this PR once https://github.com/duckdb/duckdb/pull/8774 is merged

### CODE REVIEW

**Comment by Mause** [comment]
TL;DR: we were previously setting `default_connection` to null to fix the GC reference loop, but not actually waiting for the connection to close, so users saw a wait when writing back to disk

**Review by Maxxen — APPROVED** [review]
LGTM, surprised there isn't a first party `FindNode.cmake`, although locating the headers through `NODE_ROOT_DIR` in combination with some good defaults seems fine.

**Comment by Mause** [comment]
> LGTM, surprised there isn't a first party `FindNode.cmake`, although locating the headers through `NODE_ROOT_DIR` in combination with some good defaults seems fine.

I actually just restored it exactly as it was previously written, I'll have a look to see if cmake does support it now though

**Comment by Mytherin** [comment]
Thanks for the PR! LGTM - could you just merge with master? I just merged https://github.com/duckdb/duckdb/pull/8774

File: tools/nodejs/test/close_hang.test.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import fs from "fs-extra";
+import path from "path";
+import {Database, OPEN_READWRITE} from "..";
+import {promisify} from "util";
+import {expect} from "chai";
+
+const exec = promisify(_exec);
+
+it("close hang", async function main() {
+  if (process.platform == 'win32') this.skip();
```
Comment (AlexKAparavi) [inline tools/nodejs/test/close_hang.test.ts:12]: Am I correct, that in windows it is still not resolved?

Comment (Mause) [reply tools/nodejs/test/close_hang.test.ts:12]: It's just this test that can't run on windows

Comment (AlexKAparavi) [reply tools/nodejs/test/close_hang.test.ts:12]: Yep, I see that. I'm just wondering about the reasoning behind that. As this issue is reproducing on my Windows for version 0.9.2, so I was researching if it is about me to do smth wrong. Or the issue is not fixed for Windows.

Comment (Mause) [reply tools/nodejs/test/close_hang.test.ts:12]: The issue is fixed for windows. This test uses lsof, which is is not available on windows
